packages/babel-generator/src/generators/types.js

```diff
@@ -124,7 +124,7 @@ export function NumericLiteral(node: Object) {
 
 export function StringLiteral(node: Object, parent: Object) {
   const raw = this.getPossibleRaw(node);
-  if (!this.format.minified && raw != null) {
+  if ((t.isDirectiveLiteral(node) || !this.format.minified) && raw != null) {
```

Comment:
Thanks for reviewing.  I've simplified as suggested, and also reversed order of checks to do the lightweight minified check first.